### ISSUE TITLE
Genbook chapter/section selector needs improvement

### ISSUE THREAD

**Issue by tuomas2** [mjdenham/and-bible#32]
If a book has deep and long tree structure, current chapter selector window becomes pretty much unusable as it renders all the entries of a tree structure as a flat list. 

Some tree structured selector should be implemented, or similar "two/three click" selector which is used with bibles could be also OK.

**Comment by Emet-like-AndBible** [comment]
I agree completely, I wait it already 1.5 year, but Martin while engages in more global questions...
If you will realize, will be great. DREAM!
